SimTracker/SiPhase2Digitizer/plugins/PSPDigitizerAlgorithm.h

```diff
@@ -16,9 +16,11 @@ class PSPDigitizerAlgorithm : public Phase2TrackerDigitizerAlgorithm {
 
   bool select_hit(const PSimHit& hit, double tCorr, double& sigScale) const override;
   bool isAboveThreshold(const DigitizerUtility::SimHitInfo* hitInfo, float charge, float thr) const override;
+  bool isInBiasRailRegion(const PSimHit& hit) const;
 
 private:
   edm::ESGetToken<SiPhase2OuterTrackerLorentzAngle, SiPhase2OuterTrackerLorentzAngleSimRcd> siPhase2OTLorentzAngleToken_;
   const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> geomToken_;
+  bool addBiasRailInefficiency_{false};
```

Comment:
I may be missing something but If you set it here to `false` and then you ask for the parameter to be in the config ([here](https://github.com/cms-sw/cmssw/pull/37170/files#diff-26a87b0dbe19d35bdcb12ad18825437a0c2e628498c15d0f6efdb95d8d908248R28)) this `false` will be always overwritten by the input parameter, no? If the intention was to have a default value for this (without having to specify it in the python configuration) you would need to do something like:

```c++
SPDigitizerAlgorithm::PSPDigitizerAlgorithm(const edm::ParameterSet& conf, edm::ConsumesCollector iC)
     : Phase2TrackerDigitizerAlgorithm(conf.getParameter<ParameterSet>("AlgorithmCommon"),
                                       conf.getParameter<ParameterSet>("PSPDigitizerAlgorithm"),
                                       iC),
       geomToken_(iC.esConsumes()) {
      if (conf.getParameter<ParameterSet>("PSPDigitizerAlgorithm").exists("AddBiasRailInefficiency"))
      {
       addBiasRailInefficiency_ = conf.getParameter<ParameterSet>("PSPDigitizerAlgorithm").getParameter<bool>("AddBiasRailInefficiency");
      }
```
(note that this compiles but may be prone to errors.)

To have it set to `false` if the parameter is not in the config.

Reply:
[ P.s. I thought to have already commented here but actually I had not pushed "Submit the review". ]

Reply:
@AdrianoDee  is correct - there is no sense in this construction. This should be done in utilities and not for each specific parameters. For me it is much more clear when there a python fragment responsible for parameters of this class with default values of all parameters.

Reply:
> This may be fixed in the new PR
Thanks @civanch, yes I agree (@suchandradutta).

Reply:
Indeed you are right. Still I would think that `false` is useless but I don't know how to parse a default parameter in this context (usually I go through a `fillDescriptions` but I'm not sure about how it may be implemented here).

Reply:
I guess one would need a `fillPSetDecription` in the mother class that is then percolated to the individual `plugin`s making use of the algorithm. I think this would need some larger refactoring of code (what about opening an issue?)